File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.java
class Solution {
    public long countInterestingSubarrays(List<Integer> nums, int modulo, int k) {
         int n = nums.size();
        long[] ps = new long[n + 1];
        Map<Long, Long> frq = new HashMap<>();

        ps[0] = 0;
        frq.put(0L, 1L);
        long ans = 0;

        for (int i = 0; i < n; i++) {
            ps[i + 1] = (ps[i] + (nums.get(i) % modulo == k ? 1 : 

0)) % modulo;
            long xp = (ps[i + 1] - k + modulo) % modulo;
            ans += frq.getOrDefault(xp, 0L);
            frq.put(ps[i + 1], frq.getOrDefault(ps[i + 1], 0L) + 1);
        }

        return ans;
    }
}

File: 2915-count-of-interesting-subarrays/2915-count-of-interesting-subarrays.py
class Solution:
    def countInterestingSubarrays(self, nums: List[int], modulo: int, k: int) -> int:
        n = len(nums)
        ps = [0] * (n + 1)
        frq = {}

        ps[0] = 0
        frq[0] = 1
        ans = 0

        for i in range(n):
            ps[i + 1] = (ps[i] + (nums[i] % modulo == k)) % modulo
            xp = (ps[i + 1] - k + modulo) % modulo
            ans += frq.get(xp, 0)
            frq[ps[i + 1]] = frq.get(ps[i + 1], 0) + 1

        return ans